InstanceManager.js

```diff
@@ -557,70 +651,59 @@ class InstanceManager {
 
   /// use Current/ServersRegistered to find the corresponding
   /// instance metadata
-  resolveUUID(uuid) {
+  async resolveUUID(uuid) {
     const baseUrl = endpointToUrl(this.getAgencyEndpoint());
-    return this.rpAgency({
+    const [info] = await this.rpAgency({
       method: 'POST',
       uri: baseUrl + '/_api/agency/read',
       json: true,
       body: [['/arango/Current/ServersRegistered']]
-    }).then(([info]) => {
-      const servers = info.arango.Current.ServersRegistered;
-      let url = servers[uuid].endpoint;
-      return this.instances.filter(inst => inst.endpoint == url).shift();
     });
+    const servers = info.arango.Current.ServersRegistered;
+    let url = servers[uuid].endpoint;
+    return this.instances.filter(inst => inst.endpoint == url).shift();
   }
 
-  assignNewEndpoint(instance) {
+  async assignNewEndpoint(instance) {
     let index = this.instances.indexOf(instance);
```

Comment:
Nitpick: we should inline `index` in both cases where we use it in this function. The declaration makes it look like it's going to be used later but in both cases it's just used in the immediately next line and then reassigned before being used again.